### DISCUSSION TITLE
immer wieder der Error - Rate too high

### DISCUSSION THREAD

**Discussion by pimp1310** [jomjol/AI-on-the-edge-device#2261] [Q&A]
Hallo,

ich bin erst seit 4 tagen bei dem Projekt aber habe leider auch ein paar Probleme, und zwar erhalte ich immer wieder den Fehler hier mal der letzte.

`Rate too high - Read: 339.6995 - Pre: 339.4964 - Rate: 0.2031`

da kann ich ja nur die Previous Value zurücksetzten oder?
wie genau ist da die beste Vorgehensweise um den Fehler zu beseitigen?

Meine installierte Version

![main](https://user-images.githubusercontent.com/6306057/228282427-93011195-252b-4fda-92a7-a9264b16fe1a.jpg)

hier sind meine Settings 

![settings1](https://user-images.githubusercontent.com/6306057/228281520-189baae8-d93c-4919-bbf5-1f7039f03096.jpg)
![settinghs2](https://user-images.githubusercontent.com/6306057/228281704-b5d9c869-0263-48e6-9408-575b1db978bf.jpg)
![settings4](https://user-images.githubusercontent.com/6306057/228281847-8e9a5ca2-ee6e-4a28-b45f-d4e20340b8f4.jpg)

mir wurde empfohlen die einstellung "Maximum Rate Type" zu deaktivieren, ist das korrekt?

Danke

**Comment by friedpa** [comment]
Was war denn der Wert den man ablesen konnte. Oder poste einfach das Bild dazu.
Nochwas: In welchem Zeitabstand wird gemessen und welche Wassermenge kann in der Zeit durchfließen (ist Wasserzähler abhängig). Wenn dann die MaxRate zu niedrig eingestellt wird, kann es auch zu einer Überschreitung des Wertes kommen. Bei meinem Wassermeter steht der MaxRateTyp auf RateChange und die Rate Value auf 0,07 bei einem 3 Minuten Scanzyklus

**Reply by pimp1310** [reply]
hi, 

ja ich habe es wieder 50minuten zu spät gesehen, aber das ist das aktuelle 

![Unbenannt](https://user-images.githubusercontent.com/6306057/228295698-06aec060-3cdc-4842-abab-43c3b8d9f694.jpg)


in 5 Minuten abständen wird gemessen, so laut meiner frau gehen da 2500kubik/h max durch.

**Reply by caco3** [reply]
2500m3/h?
Betreibt ihr ein Hallenbad?

Bei uns gehen an einem Waschtag max 400l/Tag durch! Und auch kurzzeitig nie so viel wie du angibst!

**Reply by pimp1310** [reply]
ne ich dachte du fragst was der zähler kann :-)
das würde passen mit 400l/Tag

**Reply by pimp1310** [reply]
ich habe deine werte mal probiert, aber hatte dann auch sofort den error diesmal aber einen anderen

Neg. Rate - Read: - Raw: 0340.2056 - Pre: 340.9991

**Comment by friedpa** [comment]
Bei mir hat das Setzen von Check digit Increase Consistency auf true geholfen (Poste einmal Deine PostProcessing Paramater aus der Expert Einstellung)

**Reply by pimp1310** [reply]
![expert3](https://user-images.githubusercontent.com/6306057/228305032-a4b84fd2-d439-4fd3-b67e-bee31c3fe28b.jpg)

und er ist wieder in den error gelaufen

**Comment by friedpa** [comment]
Also hast Du Check Digit Increase Consistency auf true gesetzt? Und dann einen Reset gemacht. Das Problem was ich oben erkennen kann ist der 0 Durchlauf. In dem Bild oben steht der Wert auf false
Nachtag: das sieht mir wie ein BM Q3 4 Zähler aus: also 4m3/h -> in 5 minuten können also max 333,3 Liter fließen, der Wasserzähler läuft aber nie voll, das hat mich in meinem Fall auf die 70l in 3 Minuten gebracht, also in Deinem Fall circa 0,120 als MaxRate oder eben 120l in 5 Minuten. Dein Problem ist aber der 0 Durchlauf

**Reply by pimp1310** [reply]
ja der steht jetzt auf true. mit reset meinst du reboot?
soll ich die werte mal einstellen? MaxRate =  0,120 ? und 5 min ist ja schon eingestellt.
ja das habe ich auch schon vermutet mit dem nulldurchlauf. was kann man da machen?

**Comment by friedpa** [comment]
Check Digit Increase Consistency auf true stellen und einen reboot machen

**Reply by pimp1310** [reply]
ja habe ich doch schon steht ja eins drüber :-)

**Comment by friedpa** [comment]
Sorry, nicht gesehen. Was ist die jetzige Fehlermeldung, wenn es eine gibt?

**Reply by pimp1310** [reply]
Neg. Rate - Read: - Raw: 0340.2988 - Pre: 340.3088

gerade erhalten

**Comment by friedpa** [comment]
Bitte beobachte jetzt wie oft das negativ rate jetzt auftritt, das ist bei mir auch der Fall aber über den Tag mittelt es sich aus. Sonst würde vielleicht ein anderes tflite Modell helfen, da kenne ich mich aber zuwenig aus,

**Reply by pimp1310** [reply]
misst er denn trotz des errors weiter? ich glaube der steckt dann fest das hatte ich eben schon mal.

**Reply by pimp1310** [reply]
oder ist meine Kalibrierung fehlerhaft ?


![calibration1](https://user-images.githubusercontent.com/6306057/228338389-3fa8784d-2645-4459-b6e0-62e155247265.jpg)
![calibration2](https://user-images.githubusercontent.com/6306057/228338420-6682f492-a675-44cb-821c-c2cbc9f1040e.jpg)

**Comment by friedpa** [comment]
Die Digits sind nicht gut, schau Dir im Wiki das richtige setting an

**Comment by ghost** [comment]
Achte mal darauf wann sich das kleinste Digit anfängt umzustellen. Dann trage den Wert des größten Zeigers in `Analog/Digital Transition Start` ein. Bei dir vermutlich 8.5 oder so.

**Reply by pimp1310** [reply]
deine Antwort habe ich jetzt erst gesehen, der steht aktuell auf 9,2.
wie meinst du das genau? wenn digit 4 die zahl wechselt?

**Comment by pimp1310** [comment]
so ich hatte den **Maximum Rate Value** auf 0.240 gestellt, davor auf 0.120 aber er läuft wieder in den fehler mit dem negativ 
`Neg. Rate - Read: - Raw: 0340.6014 - Pre: 340.6115`

das ist doch weil er die analogen nicht richtig erkennt oder? die zahlen nach dem Komma stimmen ja nicht , so wie ich das verstehe. Was könnte ich noch machen?

**Reply by caco3** [reply]
zeig mal die Recognition page und versuchs mit einem anderen analog model

**Reply by pimp1310** [reply]
![reconigtion](https://user-images.githubusercontent.com/6306057/228436269-b58cd12e-0e11-4863-8491-eda43fde163c.jpg)

habe ich probiert, habe ja nur zwei zur auswahl bei den analogen.

**Reply by caco3** [reply]
Die recognition passt ja eigentlich, er erkennt da gerade alles richtig.

Ev. kannst Du den Fokus noch etwas besser machen?
Und den Kontrast?

Dein Sprung war ja beim zweiten Zeiger, also nicht beim Übergang zum Digit, daher kann `Analog/Digital Transition Start` da nicht helfen.

Dass er hin und wieder mal etwas falsch liest, ist normal und bedingt durch den Neueronalen Netzwerk-Mechanismus.
Ich hae das auch hin und wieder.
Das kannst Du abfedern, indem Du `max rate` sinnvoll eingrenzt.
Bei mir steht das z.b. 0.05 (pro 2 Minuten).

**Reply by pimp1310** [reply]
ja mit dem fokus da habe ich schon lange probiert, das war so das beste was ich rausholen konnte glaube ich, ist das zu unscharf?
klar mit dem kontrast könnte ich noch spielen, habe ich bisher noch gar nicht.

ja die max rate hatte ich schon langsam hoch gestelllt oder wie hier empfohlen mal auf 0.120 gestellt, das reicht auch nicht mittlerweille bin ich bei 0.240, ist das zu hoch bei 5 Minuten taktung ?

**Reply by caco3** [reply]
Du könntest mit der Kamera noch etwas näher ran, bei mir ist das z.b. so:
![grafik](https://user-images.githubusercontent.com/1783586/228442221-f1a28d1a-031d-41ed-ab0c-050be4da7d7b.png)

Die rate must Du so einstellen, dass es euren normalen verbauch nicht überschreitet. Sie dient ja dazu, allfällige false readings abzufangen.
es kann dann sin, dass Du mal ein rate to high hast, aber 5 minuten später passt es dann ja wieder.
umgekehrt ist es mühsammer, wenn er zu neg rate hat, musst du warten, bis er diese 100l oder so verbraucht hat.

**Reply by pimp1310** [reply]
ja okay das ist schicker, und schärfer.

wie berechne ich die denn richtig ? sagen wir maximal 400l am tag wegen waschmaschine und spülmaschine sowie baden usw. bei einem 3 minuten zyclus

**Reply by caco3** [reply]
der Tagesverbrauch ist nicht relevant. Wichtig ist der maximal mögliche Verbrauch pro Zeitinterval, also 5 Min bei dir.

Schau dir mal den Data Graph an. Bei mir ist da die maximale Rate (pro 2 Minuten) auf ca. 0.01.

![grafik](https://user-images.githubusercontent.com/1783586/228456809-8d7f1903-f68a-4921-b4ee-89a9efe8710a.png)

Also stelle ich die max. rate etwas darüber ein. Ich habs bei mir bei 0.05, aber 0.02 müssten auch reichen, ev. sogar 0.015.

**Reply by pimp1310** [reply]
okay danke dafür muss erstmal etwas auslesen über einen längeren zeitraum, jetzt hatte ich wieder den negativ rate error, und dies liegt er wirklich daneben 

![Unbenannt](https://user-images.githubusercontent.com/6306057/228507905-a62e848c-edbc-4850-a631-820cd6600ad6.jpg)

**Reply by caco3** [reply]
hmm, die 5.3 sind komisch. ev. liegt es am kontrast, sonst sehe ich auch nicht, was da falsch läuft.

**Reply by pimp1310** [reply]
die 7.3 auch´, das sind ja 8,3, habe den kontrast jetzt erhöht auf 20. ich finde das bild gar nicht schlecht oder ? habe die kamera auch näher ran gesetzt.

![reconigtion](https://user-images.githubusercontent.com/6306057/228514536-1a0b414d-ea71-4cdf-bb8a-ceaf6c4afc13.jpg)

**Comment by friedpa** [comment]
Habe ich eh oben versucht zu erklären: das sieht mir wie ein BM Q3 4 Haushaltszähler aus: also max 4m3/h -> in 5 Minuten können also max 333,3 Liter fließen, der Wasserzähler läuft aber nie voll, das hat mich in meinem Fall auf die 70l in 3 Minuten gebracht, also in Deinem Fall circa 0,120 als MaxRate oder eben 120l in 5 Minuten.

**Reply by pimp1310** [reply]
ja stimmt, danke dir nochmals, aktuell habe ich wieder nen negativ rate error, ich packe die kamera mal näher ran an den zähler

**Comment by pimp1310** [comment]
wieder der error Neg. Rate - Read: - Raw: 340.8044 - Pre: 340.8091

ich kriege das nicht in den griff

**Comment by friedpa** [comment]
Du musst hier wirklich in Betracht ziehen, dass das System öfters "fehlinterpretiert" Im LOG siehst Du nur die Fehler, die "gültigen" Messungen werden im LOG nicht angezeigt..... 
Hier ein Auszug aus meinem LOG von heute (Gaszähler) beim Wasserzähler mit seinen Analogzeigern habe ich noch viel mehr Einträge)

[4d13h27m27s] 2023-03-29T07:14:38 <ERR> [POSTPROC] default: Raw: 22143.99, Value: , Status: Rate too high - Read: 22143.99 - Pre: 22143.41 - Rate: 0.19
[4d13h33m27s] 2023-03-29T07:20:38 <ERR> [POSTPROC] default: Raw: 22145.60, Value: , Status: Rate too high - Read: 22145.60 - Pre: 22143.55 - Rate: 0.68
[4d14h33m27s] 2023-03-29T08:20:37 <ERR> [POSTPROC] default: Raw: 22144.99, Value: , Status: Rate too high - Read: 22144.99 - Pre: 22144.57 - Rate: 0.14
[4d16h36m27s] 2023-03-29T10:23:37 <ERR> [POSTPROC] default: Raw: 22145.35, Value: , Status: Neg. Rate - Read: - Raw: 22145.35 - Pre: 22145.38
[4d16h39m27s] 2023-03-29T10:26:37 <ERR> [POSTPROC] default: Raw: 22145.36, Value: , Status: Neg. Rate - Read: - Raw: 22145.36 - Pre: 22145.38
[4d17h42m27s] 2023-03-29T11:29:37 <ERR> [POSTPROC] default: Raw: 22145.90, Value: , Status: Neg. Rate - Read: - Raw: 22145.90 - Pre: 22145.94
[4d19h00m26s] 2023-03-29T12:47:37 <ERR> [POSTPROC] default: Raw: 22146.99, Value: , Status: Rate too high - Read: 22146.99 - Pre: 22146.38 - Rate: 0.20
[4d20h36m26s] 2023-03-29T14:23:36 <ERR> [POSTPROC] default: Raw: 22147.30, Value: , Status: Neg. Rate - Read: - Raw: 22147.30 - Pre: 22147.31
[4d20h39m26s] 2023-03-29T14:26:36 <ERR> [POSTPROC] default: Raw: 22147.30, Value: , Status: Neg. Rate - Read: - Raw: 22147.30 - Pre: 22147.31
[4d20h42m26s] 2023-03-29T14:29:36 <ERR> [POSTPROC] default: Raw: 22147.30, Value: , Status: Neg. Rate - Read: - Raw: 22147.30 - Pre: 22147.31
[4d20h54m26s] 2023-03-29T14:41:36 <ERR> [POSTPROC] default: Raw: 22147.99, Value: , Status: Rate too high - Read: 22147.99 - Pre: 22147.44 - Rate: 0.18
[4d21h33m26s] 2023-03-29T15:20:36 <ERR> [POSTPROC] default: Raw: 22147.74, Value: , Status: Neg. Rate - Read: - Raw: 22147.74 - Pre: 22147.78
[4d21h54m27s] 2023-03-29T15:41:37 <ERR> [POSTPROC] default: Raw: 22147.94, Value: , Status: Neg. Rate - Read: - Raw: 22147.94 - Pre: 22147.98

**Comment by ghost** [comment]
Da der Fehler nicht bei der ersten Ziffer ist, sondern zwischen den Zeigen, liegt das vermutlich wirklich an der KI. Du verwendest ein dig-cont und vermutlich auch ein ana-cont. Einfach mal ein ana-class100 verwenden oder eine andere Version vom dig-cont. 

Leider haben wir bei den analogen Zeigern nur 6000 Bilder. Hört sich viel an, ist aber nicht viel. Bei den 20000 Zifferbildern merkt man schon einen Unterschied. Da ist die Ungenauigkeit nicht so groß.

Wir nehmen gerne weitere Bilder an https://github.com/haverland/collectmeteranalog

**Reply by pimp1310** [reply]
hi, 

bei den digitalen verwende ich "dig-cont_0611_s3" und bei den analogen "ana-class100_0157_s1_q" , ich habe aber bereits beide verfügbaren bei den analogen probiert.
gibt es da noch andere zum download? 

Das problem sind doch nur die analogen oder verstehe ich das falsch? weil du sagtest eine andere version des dig-cont.


Ich habe jetzt nochmals ne neue halterung gedruckt und bin wesentlich näher an die Uhr ran, und habe mit dem kontrast usw gespielt das hier ist das ergebnis ich denke das ist besser wie zuvor.

![reference](https://user-images.githubusercontent.com/6306057/228631099-e2986844-fc9a-4515-9886-081b43ef7e23.jpg)

![reference2](https://user-images.githubusercontent.com/6306057/228631889-fdc4c171-4152-44e2-965d-e3186a90773f.jpg)

**Reply by ghost** [reply]
Das class-157 war nicht so toll. Es ist overfittet. Heißt es kann mit anderen Bildern nicht gut umgehen. Das class100-158 und das dig-cont-1203 sind besser. Wobei das dig-cont-1203 noch recht neu ist. Es funktioniert gut mit meinem Wasserzähler, aber ich habe auch viele eigene Bilder beigesteuert.

https://github.com/jomjol/neural-network-analog-needle-readout

**Reply by pimp1310** [reply]
ja okay, ich habe da ja keine wirkliche Erfahrung mit, also verzeih mir dumme fragen, nehme ich die Datei "ana-class100_0158_s1_q.tflite" https://github.com/jomjol/neural-network-analog-needle-readout/blob/main/ana-class100_0158_s1_q.tflite und füge die auf die sd karte ein? wohin muss die ?

 das "dig-cont-1203" finde ich nicht unter dem link?

**Reply by caco3** [reply]
Du kannst sie direkt via Webinterface hochladen: Menu System-File Server.
Dann in den Ordner config wechseln

**Reply by pimp1310** [reply]
ah okay danke, wo finde ich denn noch die erwähnte " dig-cont-1203" ?

**Reply by pimp1310** [reply]
hmm finde die "dig-cont-1203" nirgends

**Comment by ghost** [comment]
Deine Schärfe der Bilder ist übrigens inzwischen wirklich gut. Das sollte helfen, eine bessere Erkennung zu erreichen. Dein zweiter Zeiger ist etwas verschoben zum Ersten. 4.6 oder 4.7 beim Ersten. Da würde ich erwarten, dass der zweite Zeiger zwischen 6 und 7 steht.

PS: Das sollte aber keinen Einfluss auf das korrekte Ergebnis haben.

**Reply by pimp1310** [reply]
kannst mir noch sagen wo ich die "dig-cont-1203" finde?

**Reply by ghost** [reply]
https://github.com/jomjol/neural-network-analog-needle-readout/blob/main/ana-cont_1203_s2_q.tflite

**Comment by pimp1310** [comment]
![reference](https://user-images.githubusercontent.com/6306057/228815984-97b1f2ed-2419-4833-a595-4b498c0aeefe.jpg)


auf dem bild hier das sind doch keine 340,9319 dann müsste sich der digi ja auch bewegen die 0 auf die 1.
da ist der analog 1 mit 2 vertauscht

**Reply by ghost** [reply]
Die Reihenfolge deiner Zeiger hat nichts mit den Namen zu tun. Prüfe mal ob die Reihenfolge auch den entsprechenden Zeigern  zugehörig ist:
<img width="465" alt="image" src="https://user-images.githubusercontent.com/412645/228899521-c2b7175f-7c99-4fbf-b8f0-55287bfd0de6.png">

**Reply by ghost** [reply]
Ist im 1. Bild schon zu sehen. Zeiger1 und Zeiger2 sind vertauscht.

**Comment by pimp1310** [comment]
so dank eurer unermütlichen Hilfe und Unterstützung funktioniert alles nun einwandfrei, ich habe zwar ab und zu eine fehlmessung aber zu 98% sind jetzt alle Messung OK.

Ich bedanke mich nochmals bei allen die mir geholfen habe, und danke für dieses super Projekt!!
